src/main/java/com/microsoftopentechnologies/windowsazurestorage/AzureStorageBlueArtifact.java

```diff
@@ -0,0 +1,69 @@
+package com.microsoftopentechnologies.windowsazurestorage;
+
+import hudson.Extension;
+import hudson.model.Run;
+import io.jenkins.blueocean.rest.Reachable;
+import io.jenkins.blueocean.rest.factory.BlueArtifactFactory;
+import io.jenkins.blueocean.rest.hal.Link;
+import io.jenkins.blueocean.rest.model.BlueArtifact;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public final class AzureStorageBlueArtifact extends BlueArtifact {
+    private final AzureBlobAction action;
+    private final AzureBlob artifact;
+
+    public AzureStorageBlueArtifact(AzureBlobAction action, AzureBlob artifact, Link parent) {
+        super(parent);
+        this.action = action;
+        this.artifact = artifact;
+    }
+
+    @Override
+    public String getName() {
+        return artifact.getBlobName();
+    }
+
+    @Override
+    public String getPath() {
+        return artifact.getBlobName();
+    }
+
+    @Override
+    public String getUrl() {
+        return String.format("/job/%s/%d/Azure/processDownloadRequest/%s", action.getContainerName(),
+                action.getBuild().number, artifact.getBlobName());
```

Comment:
Do we only support azure `blob` storage? Do we need to take the `share` storage into consideration?

Reply:
It seems that all items are stored in `individualBlobs` as `AzureBlob`, and `AzureBlob` has a variable `strogeType` to distinguish from `blob` to `share`.

Reply:
I mean, you are constructing the url using ` action.getContainerName()` which I assume only applies to `blob` storage.  `containeName` of `AzureBlobAction` should be empty in case of `share` storage.

Reply:
Fixed.

Reply:
Great, please move forward.